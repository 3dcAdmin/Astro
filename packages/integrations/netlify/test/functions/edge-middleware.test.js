import netlifyAdapter from '../../dist/index.js';
import { testIntegration, loadFixture } from './test-utils.js';
import { expect } from 'chai';
import chaiJestSnapshot from 'chai-jest-snapshot';

describe('Middleware', () => {
	it('with edge handle file, should successfully build the middleware', async () => {
		/** @type {import('./test-utils').Fixture} */
		const fixture = await loadFixture({
			root: new URL('./fixtures/middleware-with-handler-file/', import.meta.url).toString(),
			output: 'server',
			adapter: netlifyAdapter({
				dist: new URL('./fixtures/middleware-with-handler-file/dist/', import.meta.url),
			}),
			site: `http://example.com`,
			integrations: [testIntegration()],
			build: {
				excludeMiddleware: true,
			},
		});
		await fixture.build();
		const contents = await fixture.readFile('../.netlify/edge-functions/edgeMiddleware.mjs');
		expect(contents.includes('title:')).to.be.true;
		chaiJestSnapshot.setTestName('Middleware with handler file');
		expect(contents).to.matchSnapshot(true);
	});

	it('without edge handle file, should successfully build the middleware', async () => {
		/** @type {import('./test-utils').Fixture} */
		const fixture = await loadFixture({
			root: new URL('./fixtures/middleware-without-handler-file/', import.meta.url).toString(),
			output: 'server',
			adapter: netlifyAdapter({
				dist: new URL('./fixtures/middleware-without-handler-file/dist/', import.meta.url),
			}),
			site: `http://example.com`,
			integrations: [testIntegration()],
			build: {
				excludeMiddleware: true,
			},
		});
		await fixture.build();
		const contents = await fixture.readFile(
			// this is abysmal...
			'../.netlify/edge-functions/edgeMiddleware.mjs'
		);
		expect(contents.includes('title:')).to.be.false;
		chaiJestSnapshot.setTestName('Middleware with handler file');
		expect(contents).to.matchSnapshot(true);
	});
});
