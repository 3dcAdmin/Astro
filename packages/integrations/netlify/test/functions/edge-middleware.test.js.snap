// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Middleware with handler file 1`] = `
"// test/functions/fixtures/middleware-with-handler-file/.netlify/edge-functions/edgeMiddleware.mjs
import { createContext, trySerializeLocals } from \\"astro/middleware\\";
globalThis.process = {
  argv: [],
  env: Deno.env.toObject()
};
function netlify_edge_middleware_default({ request, context }) {
  return {
    title: \\"Hello world\\"
  };
}
var onRequest = (context, next) => {
  context.locals.title = \\"Middleware\\";
  return next();
};
async function middleware(request, context) {
  const url = new URL(request.url);
  const ctx = createContext({
    request,
    params: {}
  });
  ctx.locals = netlify_edge_middleware_default({ request, context });
  const next = async () => {
    const response = await fetch(url, {
      headers: {
        \\"x-astro-locals\\": trySerializeLocals(ctx.locals)
      }
    });
    return response;
  };
  return onRequest(ctx, next);
}
export {
  middleware as default
};
"
`;

exports[`Middleware with handler file 2`] = `
"// test/functions/fixtures/middleware-without-handler-file/.netlify/edge-functions/edgeMiddleware.mjs
import { createContext, trySerializeLocals } from \\"astro/middleware\\";
globalThis.process = {
  argv: [],
  env: Deno.env.toObject()
};
var onRequest = (context, next) => {
  context.locals.title = \\"Middleware\\";
  return next();
};
async function middleware(request, context) {
  const url = new URL(request.url);
  const ctx = createContext({
    request,
    params: {}
  });
  ctx.locals = {};
  const next = async () => {
    const response = await fetch(url, {
      headers: {
        \\"x-astro-locals\\": trySerializeLocals(ctx.locals)
      }
    });
    return response;
  };
  return onRequest(ctx, next);
}
export {
  middleware as default
};
"
`;
